Lib/asyncio/streams.py

```diff
@@ -575,6 +575,11 @@ def feed_data(self, data):
                         'Separator is not found, and chunk exceed the limit',
                         offset)
 
+            # All data that was in the buffer has been returned, if there is
+            # an exception dont wait and just raise the exception.
+            if self._exception is not None:
+                raise self._exception
+
```

Comment:
@1st1 You raised a comment in the previous MR that you didn't like it that the exception might be hidden by the `IncompleteReadError` exception. 

I've moved the check before the `IncompleteReadError` exception to get rid of that problem and make the behavior consistency through all of the read methods, so in case of there is not enough data to be returned and if there is an exception always first the Exception.

PD: under the hood if the `set_exception` is called, by design the `eof` can't be called [1]. But yes, it's an implementation detail that could be different on other platforms.

[1] https://github.com/python/cpython/blob/master/Lib/asyncio/selector_events.py#L691